src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+type Props = {
+  todo: Todo;
+  onCloseModal: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onCloseModal }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId).then(userFromServer => {
+      setUser(userFromServer);
+      setIsLoading(false);
+    });
+  }, []);
```

Comment:
- The same about loading state logic
- React Hook useEffect has a missing dependency: 'todo.userId'. Either include it or remove the dependency array